src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3604,10 +3604,26 @@ void InstanceKlass::print_on(outputStream* st) const {
     st->print("%s", source_debug_extension());
     st->cr();
   }
-  st->print(BULLET"class annotations:       "); class_annotations()->print_value_on(st); st->cr();
-  st->print(BULLET"class type annotations:  "); class_type_annotations()->print_value_on(st); st->cr();
-  st->print(BULLET"field annotations:       "); fields_annotations()->print_value_on(st); st->cr();
-  st->print(BULLET"field type annotations:  "); fields_type_annotations()->print_value_on(st); st->cr();
+  if (class_annotations() != nullptr) {
```

Comment:
I hate to say it but this whole function looks like it should be rewritten.  There are other places that could be null, like local_interfaces, and transitive_interfaces.  I wonder if you should have a macro above with a string BULLET string, and do them all like this.

Reply:
it might also be that all the metadata print_on functions should use the same thing.  There's

      static void print_value_on_maybe_null(outputStream* st, const Metadata* m) {
        if (nullptr == m)
          st->print("null");
        else
          m->print_value_on(st);
      }

maybe that should take the string with BULLET and print in the whole thing in the else statement.